lib/Runtime/PlatformAgnostic/Platform/Common/UnicodeText.Common.cpp

```diff
@@ -0,0 +1,275 @@
+//-------------------------------------------------------------------------------------------------------
+// Copyright (C) Microsoft. All rights reserved.
+// Licensed under the MIT license. See LICENSE.txt file in the project root for full license information.
+//-------------------------------------------------------------------------------------------------------
+
+#include "RuntimePlatformAgnosticPch.h"
+#include "UnicodeText.h"
+
+namespace PlatformAgnostic
+{
+namespace UnicodeText
+{
+CharacterTypeFlags GetLegacyCharacterTypeFlags(char16 ch)
+{
+    Assert(ch >= 128);
+
+    const char16 lineSeperatorChar = 0x2028;
+    const char16 paraSeperatorChar = 0x2029;
+
+    if (ch == lineSeperatorChar || ch == paraSeperatorChar)
+    {
+        return LineCharGroup;
+    }
+
+    CharacterClassificationType charClass = GetLegacyCharacterClassificationType(ch);
+
+    if (charClass == CharacterClassificationType::Letter)
+    {
+        return LetterCharGroup;
+    }
+    else if (charClass == CharacterClassificationType::DigitOrPunct)
+    {
+        return IdChar;
+    }
+    else if (charClass == CharacterClassificationType::Whitespace)
+    {
+        return SpaceChar;
+    }
+
+    return UnknownChar;
+}
+
+#if !defined(_WIN32) || ENABLE_TEST_PLATFORM_AGNOSTIC
+namespace Internal
+{
+template <typename CharType>
+inline bool isDigit(__in CharType c)
+{
+    return c >= '0' && c <= '9';
+}
+
+template <typename CharType>
+inline int readNumber(__inout CharType* &str)
+{
+    int num = 0;
+
+    // Count digits on each string
+    while (isDigit(*str))
+    {
+        num = (num * 10) + (*str - '0');
```

Comment:
Wasn't sure if overflow needed to be handled but decided that it's probably better to just do it- added a truncating overflow behavior